pkg/gameservers/gameservers.go

```diff
@@ -75,3 +75,11 @@ func applyGameServerAddressAndPort(gs *agonesv1.GameServer, node *corev1.Node, p
 
 	return gs, nil
 }
+
+// isBeforePodCreated checks to see if the GameServer is in a state in which the pod could not have been
+// created yet. This includes "Starting" in which a pod MAY exist, but may not yet be available, depending on when the
+// informer cache updates
+func isBeforePodCreated(gs *agonesv1.GameServer) bool {
```

Comment:
That a really good question!

So, reviewing this, I decided to get rid of this IsUnhealthy() implementation, because I realised that Error state should represent something catastrophic (like an invalid podspec), and shouldn't delete and restart.

My original thought was to start using gs.IsUnhealthy() all over the place - so put it everywhere where there was a check for `Unhealthy` state. But your comment made me think about this more, and I realised it wasn't a good idea.

That being said, my thought was - `isUnhealthy()` was really useful on a wider scale, whereas `isBeforePodCreated()` was only useful in this particular instance (that I could think of), so I left it in /pkg/gameservers/ .

Reply:
👍